src/private/m2tUpdater.m

```diff
@@ -27,7 +27,115 @@
+      askToShowChangelog(version);
+      reply = input(' *** Would you like to upgrade? y/n [n]:','s');
+      shouldUpgrade = strcmpi(reply(1),'y');
+      if ~shouldUpgrade
+        userInfo(verbose, ['To disable the self-updater in the future, add ' ...
+                           '"''checkForUpdates'',false" to the parameters.'] );
+      end
+  end
+end
+% ==============================================================================
+function tryToUpgrade(fileExchangeUrl, verbose)
+  % Download the files and unzip its contents into two folders
+  % above the folder that contains the current script.
+  % This assumes that the file structure is something like
+  %
+  %   src/matlab2tikz.m
+  %   src/[...]
+  %   src/private/m2tUpdater
+  %   src/private/[...]
+  %   AUTHORS
+  %   ChangeLog
+  %   [...]
+  %
+  % on the hard drive and the zip file. In particular, this assumes
+  % that the folder on the hard drive is writable by the user
+  % and that matlab2tikz.m is not symlinked from some other place.
+  pathstr    = fileparts(mfilename('fullpath'));
+  targetPath = fullfile(pathstr, '..', '..');
+
+  % Let the user know where the .zip is downloaded to
+  userInfo(verbose, 'Downloading and unzipping to ''%s'' ...', targetPath);
+
+  % Try upgrading
+  try
+      % List current folder structure. Will use last for cleanup
+      currentFolderFiles = rdirfiles(targetPath);
+
+      % The FEX now forwards the download request to Github.
+      % Go through the forwarding to update the download count and
+      % unzip
+      html          = urlread([fileExchangeUrl, '?download=true']);
+      expression    = '(?<=\<a href=")[\w\-\/:\.]+(?=">redirected)';
+      url           = regexp(html, expression,'match','once');
+      unzippedFiles = unzip(url, targetPath);
+
+      % The folder structure is additionally packed into the
+      % 'MATLAB Search Path' folder defined in FEX. Retrieve the
+      % top folder name
+      tmp          = strrep(unzippedFiles,[targetPath, filesep],'');
+      tmp          = regexp(tmp, filesep,'split','once');
+      tmp          = cat(1,tmp{:});
+      topZipFolder = unique(tmp(:,1));
+
+      % If packed into the top folder, overwrite files into m2t
+      % main directory
+      if numel(topZipFolder) == 1
+          unzippedFilesTarget = fullfile(targetPath, tmp(:,2));
+          for ii = 1:numel(unzippedFiles)
+              movefile(unzippedFiles{ii}, unzippedFilesTarget{ii})
+          end
+          % Add topZipFolder to current folder structure
+          currentFolderFiles = [currentFolderFiles; fullfile(targetPath, topZipFolder{1})];
+      end
+
+      cleanupOldFiles(currentFolderFiles, unzippedFilesTarget);
+
+      userInfo(verbose, 'Upgrade has completed successfully.');
+  catch
+      userInfo(verbose, ...
+               ['Upgrade has failed.\n', ...
+                'Please install the latest version manually from %s !'], ...
```

Comment:
I know it's unneeded, it's just a personal deviation from that rule since there is an URL before: that makes it easier to copy-paste.